src/groth16/aggregate/proof.rs

```diff
@@ -234,6 +234,12 @@ impl<E: Engine> GipaProof<E> {
         let mut buffer = 0u32.to_le_bytes();
         source.read_exact(&mut buffer)?;
         let nproofs = u32::from_le_bytes(buffer);
+        if nproofs < 2 || !nproofs.is_power_of_two() {
```

Comment:
* This would need to be present in the `verify_aggregate` function -
* It's an added feature, so I'd prefer to do it in another PR if we really wan to do it ?
* "0 proofs should verify" -> im not sure about that behavior, if we really want to do it, I'd prefer to default to an error as (any) application should never have to verify 0 proofs; if it does, it might be a malicious behavior (because it doesn't make sense to for the normal flow to verify 0 proofs).

Reply:
I think failing on anything less than 2 is good